src/rebar_file_utils.erl

```diff
@@ -273,6 +275,22 @@ canonical_path([_|Acc], [".."|Rest])  -> canonical_path(Acc, Rest);
 canonical_path([], [".."|Rest])       -> canonical_path([], Rest);
 canonical_path(Acc, [Component|Rest]) -> canonical_path([Component|Acc], Rest).
 
+%% returns canonical target of path if path is a link, otherwise returns path
+-spec resolve_link(string()) -> string().
+
+resolve_link(Path) ->
+    case file:read_link(Path) of
+        {ok, Target} ->
+            canonical_path(filename:absname(Target, filename:dirname(Path)));
+        {error, _} -> Path
+    end.
```

Comment:
Sounds good. I was just wondering about the configuration name for this PR
if others were going to use it.

On Tuesday, 3 May 2016, Fred Hebert <notifications@github.com> wrote:

> In src/rebar_file_utils.erl
> <https://github.com/erlang/rebar3/pull/1174#discussion_r61871296>:
>
> > @@ -273,6 +275,22 @@ canonical_path([_|Acc], [".."|Rest])  -> canonical_path(Acc, Rest);
> >  canonical_path([], [".."|Rest])       -> canonical_path([], Rest);
> >  canonical_path(Acc, [Component|Rest]) -> canonical_path([Component|Acc], Rest).
> >
> > +%% returns canonical target of path if path is a link, otherwise returns path
> > +-spec resolve_link(string()) -> string().
> > +
> > +resolve_link(Path) ->
> > +    case file:read_link(Path) of
> > +        {ok, Target} ->
> > +            canonical_path(filename:absname(Target, filename:dirname(Path)));
> > +        {error, _} -> Path
> > +    end.
>
> I would be of the opinion not to configure it. Just display the most
> helpful path there.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/erlang/rebar3/pull/1174/files/da0e2c22959c976420b60def760b348aa200d6e2#r61871296>
>